apps/mobile/src/hooks/useDelegateKey.ts

```diff
@@ -0,0 +1,117 @@
+import { useState, useCallback } from 'react'
+import { ethers } from 'ethers'
+import DeviceCrypto from 'react-native-device-crypto'
+import * as Keychain from 'react-native-keychain'
+
+import Logger from '../utils/logger'
+import { Address } from '@/src/types/address'
+import GatewayService from '../services/gateway/GatewayService'
+import { useSign } from './useSign'
+import { asymmetricKey, keychainGenericPassword } from '@/src/store/constants'
+
+export function useDelegateKey() {
+  const [loading, setLoading] = useState<boolean>(false)
+  const [error, setError] = useState<Error | null>(null)
+  const { getPrivateKey, createMnemonicAccount } = useSign()
+
+  const createDelegate = useCallback(async (ownerAddress: Address) => {
+    setLoading(true)
+    setError(null)
+
+    try {
+      if (!ownerAddress) {
+        throw Logger.info('OwnerKeyNotFoundForDelegate')
+      }
+
+      // Step 0: Get users private key from keychain. TODO: need to pass ownerAddress to getPrivateKey
+      const privateKey = await getPrivateKey()
+
+      if (!privateKey) {
+        throw Logger.error('useDelegateKey: Something went wrong', error)
+      }
+
+      // Step 1: Generate delegate key
+      const delegateMnemonic = ethers.Wallet.createRandom().mnemonic?.phrase
```

Comment:
Why do we need mnemonic here? Why don#t we go for creating a random private key?
```
const wallet = Wallet.createRandom(provider);
console.log(wallet.privatekey)
```